tgui/packages/tgui/interfaces/RobotSelfDiagnosis.js

```diff
@@ -0,0 +1,87 @@
+import { useBackend } from "../backend";
+import { NoticeBox, Flex, LabeledList, Section } from "../components";
+import { Window } from "../layouts";
+import { capitalize } from 'common/string';
+
+export const RobotSelfDiagnosis = (props, context) => {
+  const { data } = useBackend(context);
+  const { component_data } = data;
+  const getDamageColor = (damage, maxDamage) => {
+    let damageRatio = damage / maxDamage;
+    if (damageRatio <= 0.20) {
+      return "good";
+    }
+    else if (damageRatio <= 0.50) {
+      return "average";
+    }
+    else {
+      return "bad";
+    }
+  };
+  const getStatusColor = status => {
+    return status ? "good" : "bad";
+  };
```

Comment:
Perhaps move this out of the component

Reply:
Do you mean just in-line it, like `color={X ? "good" : "bad}`?

Reply:
Oh no, just moving these two functions out of the exported function, like so:

```js
const getDamageColor = ...;

const getStatusColor = ...;

export const RobotSelfDiagnosis = ...;
```